### ISSUE TITLE
Wrong lr schedule in semantic segmentation sample?

### ISSUE THREAD

**Issue by oscmansan** [pytorch/vision#2063]
Hi! I am using the semantic segmentation reference training scripts and I think I found an issue with the lr scheduler.
 
In the documentation of [torch.optim.lr_scheduler.LambdaLR](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.LambdaLR) it says that the lambda function receives an integer parameter epoch, but in the training reference script it looks that the parameter `x` is used as if it was the global step: https://github.com/pytorch/vision/blob/e61538cba036c42bab23ce8f9d205da9889977ae/references/segmentation/train.py#L158
If I understand it correctly, this is the poly learning rate policy used in [DeepLab](https://arxiv.org/pdf/1606.00915.pdf), so I think that instead it should be:
```python
lambda x: (1 - x / args.epochs) ** 0.9)
```
Also, it'd be interesting to have a way of changing the learning rate at the finer resolution of iterations, instead of epochs.

@fmassa what do you think?

**Comment by fmassa** [comment]
Hi,

Great question.

So, in https://github.com/pytorch/vision/blob/e61538cba036c42bab23ce8f9d205da9889977ae/references/segmentation/train.py#L164 you see that we pass the `lr_scheduler` to the training function, which in turn calls `step` in the training loop https://github.com/pytorch/vision/blob/e61538cba036c42bab23ce8f9d205da9889977ae/references/segmentation/train.py#L90
This means that we indeed have the correct setup, as we are changing the learning rate at the iteration resolution, and not the epochs.

So this answers both of your questions I believe. Let me know if you have other questions
